### PR TITLE
Issue #468: Display a more verbose message when GITHUB_TOKEN env variable is missing 

### LINKED ISSUE #468: Display a more verbose message when `GITHUB_TOKEN` env variable is missing

### PR DESCRIPTION
**Pull request by amalota**
closes #468

### CODE REVIEW

File: git_machete/client.py [hunk 1]
```diff
@@ -1852,12 +1852,12 @@ def checkout_github_prs(self,
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         current_user: Optional[str] = git_machete.github.derive_current_user_login()
         if not current_user and my_opened_prs:
+            msg = f"Could not determine current user name, please check your {GITHUB_TOKEN_ENV_VAR} environment variable."
```
Comment (PawelLipski) [inline git_machete/client.py]: Hmm the problem is that well, `GITHUB_TOKEN` is not the only way of providing the token... in fact this message might be confusing if someone relied on `gh`/`hub` authentication instead (and the token got expired or whatnot). Maybe just display the full(er) message as in github_api_access.rst ?

Comment (PawelLipski) [reply git_machete/client.py]: Or just redirect the user to `git machete help github` for details on how to provide the required auth!

Comment (PawelLipski) [reply git_machete/client.py]: Actually! see `git_machete.github.__fire_github_api_request`. Would be good to have a common handler code for the problems with GitHub token (the ones in `__fire_github_api_request` are slightly out of date already, `~/.github-token` are not taken into account :/)

File: git_machete/github.py [hunk 2]
```diff
@@ -165,12 +165,13 @@ def __fire_github_api_request(method: str, path: str, token: Optional[str], requ
         elif err.code in (http.HTTPStatus.UNAUTHORIZED, http.HTTPStatus.FORBIDDEN):
             first_line = f'GitHub API returned {err.code} HTTP status with error message: `{err.reason}`\n'
             if token:
-                raise MacheteException(first_line + f'Make sure that the token provided in `gh auth status` or `~/.config/hub`'
-                                                    f' or <b>{GITHUB_TOKEN_ENV_VAR}</b> is valid and allows for access to `{method.upper()}` https://{host}{path}`.')
+                raise MacheteException(first_line + f'Make sure that the GitHub API token provided by one of the: '
+                                                    f'{get_github_token_possible_providers()}is valid and allows for access to '
+                                                    f'`{method.upper()}` https://{host}{path}`.')
```
Comment (PawelLipski) [inline git_machete/github.py]: Looks that backticks are slightly confused in this line

File: git_machete/client.py [hunk 3]
```diff
@@ -1852,7 +1852,8 @@ def checkout_github_prs(self,
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         current_user: Optional[str] = git_machete.github.derive_current_user_login()
         if not current_user and my_opened_prs:
-            msg = f"Could not determine current user name, please check your {GITHUB_TOKEN_ENV_VAR} environment variable."
+            msg = ("Could not determine current user name, please check that the GitHub API token provided by one of the: "
```
Comment (PawelLipski) [inline git_machete/client.py:1855]: Could you show a screenshot of how this message looks in the terminal now?

Comment (amalota) [reply git_machete/client.py:1855]: image

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**